diff -urN xorg-cf-files-X11R7.0-1.0.1/X11.tmpl xorg-cf-files-X11R7.0-1.0.1.new/X11.tmpl
--- xorg-cf-files-X11R7.0-1.0.1/X11.tmpl	2005-10-01 18:45:48.000000000 +0200
+++ xorg-cf-files-X11R7.0-1.0.1.new/X11.tmpl	2006-04-15 14:24:40.000000000 +0200
@@ -3386,7 +3386,7 @@
 #if HasMotif
 
 #ifndef MotifDir
-#define MotifDir /usr/X11R6
+#define MotifDir /usr
 #endif
 #ifndef MotifDirStandard
 #define MotifDirStandard NO
